libraries/ms-drivers/test/test_voltage_regulator.c

```diff
@@ -0,0 +1,106 @@
+#include "adc.h"
+#include "voltage_regulator.h"
+
+#include "delay.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+#define TEST_CALLBACK_EXPECTED_TIMES_CALLED 1
+#define TEST_CALLBACK_EXPECTED_CONTEXT 0xA
+#define TEST_TIMER_CALLBACK_DELAY 1000
```

Comment:
The point of using a configurable delay is so that you can use a small delay in the tests. Currently, these tests will take at least 4 seconds to execute due to the delays. Please change this to a smaller delay like 100ms, and change the delay calls in your tests to something like `delay_ms(TEST_TIMER_CALLBACK_DELAY_MS + 10)` to give a bit of buffer for the callback to be called.

Also, please suffix this name with the unit: `TEST_TIMER_CALLBACK_DELAY_MS`.